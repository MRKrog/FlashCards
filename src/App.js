import React, { Component } from 'react';

import Header from './Header.js';
import WelcomeScreen from './WelcomeScreen.js';
import CardContainer from './CardContainer.js';
import Footer from './Footer.js';

import './styles/App.scss';

class App extends Component {
  constructor() {
    super();
    this.state = {
      currentPlayer: false,
      originCards: [],
      allCards: [],
      correctCards: [],
      currCardIndex: 0,
      localStorageExist: false,
      cardsLeft: 0
    }
  }

  componentDidMount() {
    fetch("http://memoize-datasets.herokuapp.com/api/v1/mkCards")
    .then(response => response.json())
    .then(fetchCards => {
      this.setState({
        originCards: fetchCards.mkCards,
        allCards: fetchCards.mkCards
      }, () => { this.getFromLocalStorage() });
    })
    .catch(error => {
      throw new Error(error);
    });
  }

  getFromLocalStorage = () => {
    if(localStorage.length > 0){
      const cardOrigins = this.state.allCards
      const correctArray = JSON.parse(localStorage.getItem('correctQuestions'));
      this.setState({
        correctCards: correctArray,
        localStorageExist: true
      }, () => { this.saveCardOrigins() });
    }

  }

  saveCardOrigins = () => {
    const originalCards = this.state.allCards;
    this.setState({
      originCards: originalCards
    }, () => { this.removeCorrectCards() })

  }

  removeCorrectCards = () => {
    console.log('afe', this.state.originCards);
    const { allCards, correctCards, originCards } = this.state;
    correctCards.forEach((correctCard) => {
      let foundCard = allCards.find(card => card.id === correctCard.id);
      let cardIndex = allCards.indexOf(foundCard);
      allCards.splice(cardIndex, 1);
    });
  }

  handleChangeCard = (clickedAnswer) => {
    const { allCards } = this.state;
    const currentCard = allCards[this.state.currCardIndex];
    let currIndex = this.state.currCardIndex;
    if(currentCard.correctAnswer === clickedAnswer){
      this.state.correctCards.push(currentCard);
      localStorage.setItem('correctQuestions', JSON.stringify(this.state.correctCards));
    }
    currIndex++;
    this.changeCardIndex(currIndex);
  }

  changeCardIndex = (currIndex) => {
    this.setState({
      currCardIndex: currIndex
    });
  }

  startQuiz = () => {
    this.setState({
      currentPlayer: true,
    })
  }

  resetQuiz = () => {
    localStorage.clear();
    this.setState({
      currentPlayer: false,
      localStorageExist: false,
    })

  }

  render() {
    const { allCards, currCardIndex, localStorageExist } = this.state;

    let displayContent;

    if(this.state.currentPlayer === false) {
      displayContent = <WelcomeScreen localStorageExist={localStorageExist}
                                      startQuiz={this.startQuiz} />
    } else {
      displayContent = <CardContainer currentCard={allCards[currCardIndex]}
                                      onChangeCard={this.handleChangeCard} />
    }

    console.log(this.state);
    return (
      <div className="App">
        <Header correctCards={this.state.correctCards}/>
        <main className="main-content">
          {displayContent}
        </main>
        <Footer resetQuiz={this.resetQuiz}
                currentPlayer={this.state.currentPlayer}
              />
      </div>
    )
  }
}

export default App;
