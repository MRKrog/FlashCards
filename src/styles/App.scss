
$font-title: 'Abel', sans-serif;
$font-copy: 'Lato', sans-serif;
$font-header: 'Barlow', sans-serif;

$white: #fff;
$lightWhite: #d6d6d6;
$black: #231f20;
$dark-blue: #11212a;

$purpleBtn: #723bd7;

.App {
  width: 100%;
  background:url(beach_bg.jpg);
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  z-index: 9;

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    background: rgba(76,76,76,1);
    background: linear-gradient(135deg, rgba(138,132,10,0.7) 0%, rgba(246,246,246,0.76) 47%, rgba(237,237,237,0.83) 100%);
  }

  .App-header {
    background-color: rgba(138, 132, 10, .2);
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 2em;
    z-index: 10;
    box-shadow: 0px 0px 5px 0px rgba(32, 32, 32, .5);

    h1 {
      font-family: $font-header;
      color: #bdb753;
      text-transform: uppercase;
      font-size: 1.5em;
      letter-spacing: .1em;
      margin: 0;
      line-height: 1;
      font-weight: 100;
    }

    h4 {
      font-family: $font-header;
      text-transform: uppercase;
      letter-spacing: .05em;
      color: #535667;
      font-weight: 100;
      line-height: 1;
    }

    i {
      opacity: .4
    }

  }

  .main-content {
    width: 100%;
    z-index: 10;

    .welcome-btn {
      width: 100%;
      display: block;
      margin: auto;
      max-width: 200px;
      font-family: "Lato", sans-serif;
      transition: all .5s ease;
      font-size: 1em;
      border: none;
      border-radius: 5px;
      padding: .4em;
      cursor: pointer;
      background: #1f93ec;
      color: #fff;
      outline: none;
    }
  }

  .App-footer {
    display: block;
    width: 100%;
    z-index: 10;

    p {
      margin: 0;
      text-align: center;
      color: $lightWhite;
      font-family: $font-title;
      padding: .5em
    }
  }

}
