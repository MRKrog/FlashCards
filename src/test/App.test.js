import React from 'react';
import App from '../App';
import MockData from "../mockData";
import { shallow } from "enzyme";

const mockData = MockData;

// let allInventory = mockData.distributor[0].inventory;
// const mockFunc = jest.fn();


describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have the following default state", () => {
    expect(wrapper.state()).toEqual({
      player: true,
      allCards: mockData.cards,
      correctCards: [],
      wrongCards: [],
      currCardIndex: 0
    });
  })

});
