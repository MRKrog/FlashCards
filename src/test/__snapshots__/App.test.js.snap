// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should match snapshot when all data is passed correctly 1`] = `
<div
  className="App"
>
  <Header
    correctCards={Array []}
  />
  <main
    className="main-content"
  >
    <WelcomeScreen
      localStorageExist={false}
      startQuiz={[Function]}
    />
  </main>
  <Footer />
</div>
`;
